libs/core/src/components/sage-tab/sage-tab.tsx

```diff
@@ -0,0 +1,84 @@
+import { Component, Element, Host, h, Prop, Event, EventEmitter, State } from '@stencil/core';
+
+@Component({
+  tag: 'sage-tab',
+  styleUrl: 'sage-tab.scss',
+  shadow: false,
+})
+export class SageTab {
+  @Element() el: HTMLElement;
+
+  /**
+   * Sets the related tab name, this name must match a `sage-tabpanel`'s tab property, required
+  */
+  @Prop() tab: string;
+
+  /**
+   * Keeps track of the activeTab, this property is passed in by parent component
+  */
+  @Prop({mutable: true}) activeTab: string;
+
+  /**
+   * Keeps track of the parentComponent unique id, this property is passed in by parent component
+  */
+  @Prop({mutable: true}) parentComponent: string;
+
+  /**
+   * Keeps track of if the expected tab variant, this property is passed in by parent component
+  */
+  @Prop() variant: string;
+
+  /**
+   * Keeps track of if the tabpanel is selected, this property is computed on `componentWillUpdate()`
+  */
+  @State() selected = false;
+ 
+  /**
+   * Emits an event upon tab click for `sage-tab` and `sage-tabpanel` to listen for
+   */
+  @Event() tabClick: EventEmitter<object>;
+  private onTabClick(tab, parentComponent) {
+    this.activeTab = tab;
+    this.tabClick.emit([tab, parentComponent]);
+  }
+
+  private matchActiveTab() {
+    if (this.activeTab && this.activeTab === this.tab) {
+      this.selected = true;
+    } else {
+      this.selected = false;
+    }
+  }
+
+  componentWillRender() {
+    this.matchActiveTab();
+  }
+
+  render() {
+    const availabilityTabEdgeInlineStart = (
+      <span class="sage-tab-edge" role="presentation"></span>
+    )
+
+    const availabilityTabEdgeInlineEnd = (
+      <span class="sage-tab-edge sage-tab-edge--end" role="presentation"></span>
+    )
+
+    return (
+      <Host variant={this.variant} slot="tabs">
```

Comment:
I love this idea of documenting our slots in some kind of argsTable like way. Do you know of a way to do this?

In this case, the user isn't expected to pass slot information into their `sage-tab`s or `sage-tabpanel`s but I'd still be happy to document them if we feel it'd be helpful!

Reply:
That makes sense then for this case. We can document slots the same way we do for component props and custom props. Here's an [example](https://github.com/ionic-team/ionic-framework/blob/main/core/src/components/accordion/accordion.tsx) in Ionic. It will automatically populate Storybook.

Reply:
+1 for documenting the slot name and description